cmd/node_agent/na/aws.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package na
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	cred "istio.io/auth/pkg/credential"
+)
+
+type awsPlatformImpl struct {
+	fetcher *cred.AwsTokenFetcher
+}
+
+// This is extracted from the same function in gcp.go.
+// Should put the common logic elsewhere in the future.
+func (na *awsPlatformImpl) GetDialOptions(cfg *Config) ([]grpc.DialOption, error) {
```

Comment:
I think you may need to pass the identity document to CA so that the CA can authenticate the node agent.

Reply:
True. But as I mentioned above, right now my understanding is that identity document should be passed in the `NodeAgentCredentials` field. However current implementation doesn't support overriding `na.createRequest()`. I think I'll rebase this PR once refactoring is done.

Reply:
Sounds good to me.